src/main/java/edu/illinois/cs/dt/tools/utility/iFixPlusPlugin.java

```diff
@@ -0,0 +1,744 @@
+                        return;
+                    }
+                }
+
+                for(int i=0; i<10; i++) {
+                    Try<TestRunResult> phase0ResultPass = null;
+                    try {
+                        phase0ResultPass = runner.runList(testPassOrder_full());
+                    }
+                    catch (Exception ex) {
+                        System.out.println("## Encountering error when checking the passing order: " + ex);
+                    }
+                    System.out.println("Passing order results: " +
+                            phase0ResultPass.get().results().get(dtname).result().toString());
+
+                    if (!phase0ResultPass.get().results().get(dtname).result().toString().equals("PASS")) {
+                        System.out.println("## Passing order json file is wrong!");
+                        timing(startTime);
+                        Files.write(Paths.get(output),
+                                "0,0,0,0,0,0,0,wrongjsonpass2,".getBytes(),
+                                StandardOpenOption.APPEND);
+                        return;
+                    }
+                }
+
+                timing(startTime);
+                startTime = System.currentTimeMillis();
+                System.out.println("~~~~~~ Finish checking if this test is a true order-dependent test.");
+
+                //phase 1: run doublevictim order
+                Try<TestRunResult> phase1Result = null;
+                try {
+                    System.out.println("~~~~~~ Begin running double victim order to check if it is a double victim!");
+
+                    Configuration.config().properties().
+                            setProperty("testplugin.runner.idempotent.num.runs", "2");
+                    phase1Result = runner.runList(victim());
+                    System.out.println(phase1Result.get().results().get(dtname+":1").result());
+                    Configuration.config().properties().
+                            setProperty("testplugin.runner.idempotent.num.runs", "-1");
+                    System.out.println("## Running double victim order results: " + phase1Result.get().results());
+                }
+                catch (Exception e) {
+                    System.out.println("## Encountering error when running in double victim order: " + e);
+                }
+                timing(startTime);
+                startTime = System.currentTimeMillis();
+                System.out.println("~~~~~~ Finished running double victim order!");
+
+                if (phase1Result.get().results().get(dtname+":1").result().toString().equals("PASS")) {
+                    System.out.println("~~~~~~ Begin capturing the state in passing order!");
+                    // phase 2: run doublevictim order state capture
+                    Configuration.config().properties().
+                            setProperty("statecapture.phase", "capture_after");
```

Comment:
I wonder instead of "phase" the property should actually reflect that it means the relative position for capture, maybe something like "statecapture.position"?